tenders/hvt/hvt.h

```diff
@@ -160,23 +160,26 @@ extern hvt_vmexit_fn_t hvt_core_vmexits[];
  * Module definition. (name) and (setup) are required, all other functions are
  * optional.
  */
-struct hvt_module {
-    const char *name;
+struct hvt_module_ops {
     int (*setup)(struct hvt *hvt);
     int (*handle_cmdarg)(char *cmdarg);
     char *(*usage)(void);
 };
 
-/*
- * Array of compiled-in modules. NULL terminated.
- */
-extern struct hvt_module *hvt_core_modules[];
+struct hvt_module {
+    const char *name;
```

Comment:
If you use `const char name[32];` (the compiler will complain if 32 is too small) then the modules section can contain the name like below:

 ```
[kollerr@oc6638465227 solo5]$ readelf -x modules tenders/hvt/solo5-hvt
Hex dump of section 'modules':
  0x00606180 636f7265 00000000 00000000 00000000 core............
  0x00606190 90184000 00000000 00000000 00000000 ..@.............
  0x006061a0 00000000 00000000 00000000 00000000 ................
  0x006061b0 00000000 00000000 00000000 00000000 ................
  0x006061c0 626c6b00 00000000 00000000 00000000 blk.............
  0x006061d0 202e4000 00000000 e02d4000 00000000  .@......-@.....
  0x006061e0 d02d4000 00000000 00000000 00000000 .-@.............
  0x006061f0 00000000 00000000 00000000 00000000 ................
  0x00606200 6e657400 00000000 00000000 00000000 net.............
  0x00606210 40344000 00000000 20384000 00000000 @4@..... 8@.....
  0x00606220 30324000 00000000                   02@.....
```

Reply:
Uh, yeah, the `.modules` stuff is a bit raw. I'm not going to change it now, will revisit when I get back.